### PR TITLE
doc akka http java dsl example snippets timeout directives (#20466)

### PR DESCRIPTION
**Pull request by gosubpl**
- added checks to TimeoutDirectivesExampleSpec
- added TimeoutDirectivesExamplesTest
- modified comments in Java Timeout Directives docs to include snippets
- added missing Java directive wrapper
- added working Timeout example in JavaTestServer

### CODE REVIEW

**Comment by gosubpl** [comment]
Refs #20466

**Comment by ktoso** [comment]
OK TO TEST

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala [hunk 1]
```diff
@@ -5,24 +5,70 @@
...
+import akka.http.scaladsl.model.HttpEntity._
+import akka.http.scaladsl.model._
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpec }
 import scala.concurrent.duration._
 import scala.concurrent.{ Future, Promise }
+import akka.testkit.TestKit
+
+class TimeoutDirectivesExamplesSpec extends WordSpec with Matchers with BeforeAndAfterAll with ScalaFutures with CompileOnlySpec {
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala]: You could extend AkkaSpec instead, it starts/closes actor system for you

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala [hunk 2]
```diff
@@ -5,24 +5,70 @@
...
+  implicit val system = ActorSystem(getClass.getSimpleName, testConf)
+  import system.dispatcher
+  implicit val materializer = ActorMaterializer()
+  implicit val patience = PatienceConfig(3.seconds)
+
+  override def afterAll(): Unit = {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  def slowFuture(): Future[String] = Promise[String].future
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala]: maybe add a note that on Scala 2.12 this should be `Promise.never`

**Comment by gosubpl** [comment]
PR updated. I also made the "racy" tests compile only to avoid flaky tests.

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala [hunk 3]
```diff
@@ -5,24 +5,65 @@
...
+}
+
+class TimeoutDirectivesExamplesSpec extends AkkaSpec(TimeoutDirectivesTestConfig.testConf)
+  with ScalaFutures with CompileOnlySpec {
+  //#testSetup
+  import system.dispatcher
+  implicit val materializer = ActorMaterializer()
 
-class TimeoutDirectivesExamplesSpec extends RoutingSpec with CompileOnlySpec {
+  def slowFuture(): Future[String] = Promise[String].future // move to Future.never in Scala 2.12
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala]: thanks

**Comment by ktoso** [comment]
Nice, thanks @gosubpl !
LGTM

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/TimeoutDirectivesExamplesTest.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+public class TimeoutDirectivesExamplesTest extends AllDirectives {
+    //#testSetup
+    private final Config testConf = ConfigFactory.parseString("akka.loggers = [\"akka.testkit.TestEventListener\"]\n"
+            + "akka.loglevel = ERROR\n"
+            + "akka.stdout-loglevel = ERROR\n"
+            + "windows-connection-abort-workaround-enabled = auto\n"
+            + "akka.log-dead-letters = OFF\n"
+            + "akka.http.server.request-timeout = 1000s");
+    // large timeout - 1000s (please note - setting to infinite will disable Timeout-Access header
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/TimeoutDirectivesExamplesTest.java]: I made ticket https://github.com/akka/akka/issues/20817 for it

**Comment by ktoso** [comment]
LGTM, thanks!

**Comment by gosubpl** [comment]
Many thanks @ktoso, see you!

**Pull request merged**